go/vt/vttablet/tabletserver/connpool/dbconn.go

```diff
@@ -291,7 +293,16 @@ func (dbc *DBConn) Recycle() {
 func (dbc *DBConn) Kill(reason string, elapsed time.Duration) error {
 	tabletenv.KillStats.Add("Queries", 1)
 	log.Infof("Due to %s, elapsed time: %v, killing query %s", reason, elapsed, dbc.Current())
-	killConn, err := dbc.pool.dbaPool.Get(context.TODO())
+	// Hack: Most of the times DBConn is created with a pool. There is a snowflake case
```

Comment:
Going down this route, I think you should just always set `dbaPool` for both normal and appDebug connections, so you don't need this special case logic here.

The alternative would be to always set `pool` and have something like a `noRecycle` flag in the `DBConn` to indicate if it shouldn't be returned to the pool when recycling.

Reply:
I thought about setting `dbaPool` for both normal case and appDebug,  but didn't want to change the normal flow because of this. I think that works. I would lean towards this approach.

I've tried the second alternative but founded more complex as maintaining that `notRecycle` state in the connection is not very straight forward.